server/channels/store/localcachelayer/layer_test.go

```diff
@@ -148,3 +150,28 @@ func tearDownStores() {
 		wg.Wait()
 	})
 }
+
+func TestClearCacheCluster(t *testing.T) {
+	cluster := &testlib.FakeClusterInterface{}
+	lc := &LocalCacheStore{
+		cluster: cluster,
+	}
+
+	c := cache.NewLRU(&cache.CacheOptions{
+		Size:                   10,
+		Name:                   "test",
+		InvalidateClusterEvent: model.ClusterEventInvalidateCacheForRoles,
+	})
+
+	lc.doClearCacheCluster(c)
+	assert.Len(t, cluster.GetMessages(), 1)
+
+	c = cache.NewLRU(&cache.CacheOptions{
+		Size:                   10,
+		Name:                   "test",
+		InvalidateClusterEvent: model.ClusterEventNone,
```

Comment:
How are we testing this case exactly? I don't see a difference in assertions but the `doClearCacheCluster` implementation suggests we would be skipping this case. Could we assert that somehow?

Reply:
Yes, we are asserting that we are skipping the cluster broadcast by checking the count of `cluster.GetMessages()`. If you remove the `cache.GetInvalidateClusterEvent() != model.ClusterEventNone` condition, then the test will fail.

Reply:
Oh, messages are not popped from there, I see.